### DISCUSSION TITLE
Unattended installation

### DISCUSSION THREAD

**Discussion by MarcOverIP** [Security-Onion-Solutions/securityonion#8152] [General] [answered]
Based on twitter conversations (https://twitter.com/InfosecGoon/status/1537823467613061120, https://twitter.com/securityonion/status/1537821265389658113) I was redirected and starting this discussion.

My goal: deploy SO in student lab environments in an automated way. 

Background: The labs are hosted in Azure. Everything in the labs is deployed using automation (terraform and ansible). Im looking for a way to also automate the installation of SO, without the need to create a custom SO images and redeploy that in each lab.

Challenge: I'm unaware of automation options for SO. Based on the twitter conversations I was told there is an - undocumented and subject to change - option for automated installation.

Question: where can I get more info on ways for automated installation of SO? A search for similar discussions and found [this](https://github.com/Security-Onion-Solutions/securityonion/discussions/2747). But Im wondering if this approach still applies to current SO and if it applies to the azure image.

**Comment by dougburks** [comment, marked as answer]
As mentioned on Twitter, unattended installation is undocumented and unsupported:
- undocumented meaning that you would need to look at the source code to figure out how to do this
- unsupported meaning that we don't provide any guarantees that it will work for you and we can't provide any support if you have questions or problems

If you have questions or problems other than unattended installation, please start a new discussion and we'll be glad to take a look.

**Comment by MarcOverIP** [comment]
I was hoping for a bit more :) But I understand. 

I'll make a feature request on the GitHub issue section.
